packages/webdriverio/src/commands/browser/mock.ts

```diff
@@ -100,34 +100,36 @@ import DevtoolsNetworkInterception from '../../utils/interception/devtools'
 import WebDriverNetworkInterception from '../../utils/interception/webdriver'
 import { getBrowserObject } from '../../utils'
 
-const SESSION_MOCKS: Set<Interception> = new Set()
+const SESSION_MOCKS: Map<string, Set<Interception>> = new Map()
 
-export default async function mock (this: WebdriverIO.BrowserObject, url: string, filterOptions: WebdriverIO.MockFilterOptions) {
+export default async function mock(this: WebdriverIO.BrowserObject, url: string, filterOptions: WebdriverIO.MockFilterOptions) {
     const NetworkInterception = this.isSauce ? WebDriverNetworkInterception : DevtoolsNetworkInterception
-
     if (!this.isSauce) {
         await this.getPuppeteer()
     }
-
+    const browser = getBrowserObject(this)
+    const windowHandle: string = await browser.getWindowHandle()
     /**
      * enable network Mocking if not already
      */
-    if (SESSION_MOCKS.size === 0 && !this.isSauce) {
-        const [page] = await this.puppeteer.pages()
-        const client = await page.target().createCDPSession()
-        await client.send('Fetch.enable', {
-            patterns: [{ requestStage: 'Request' }, { requestStage: 'Response' }]
+    if (!SESSION_MOCKS.has(windowHandle) && !this.isSauce) {
+        SESSION_MOCKS.set(windowHandle, new Set())
+        const pages = await this.puppeteer.pages()
+        await pages.forEach(async (page: any) => {
```

Comment:
I'd define mocks per page rather than have a mock for all pages.
It's impossible to identify where does request come from